### PR TITLE
Fix flow and eslint warnings

### PR DESCRIPTION
**Pull request by mkimberlin**

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -8,4 +9,5 @@
 
 [options]
 module.name_mapper='.*\(\.s?css\)' -> 'empty/object'
+module.name_mapper='^\(\(components\|containers\|fixtures\|middleware\|routes\|styleguide\|styles\|util\)/.*\)$' -> '<PROJECT_ROOT>/src/\1'
```
Comment (mkimberlin) [inline .flowconfig:12]: This is annoying, but the only way I can find to get Flow to recognize the relative module imports...

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.*
+package.json
+package-lock.json
+*.md
+*.lock
```
Comment (mkimberlin) [inline .prettierignore:5]: make prettier stop formatting some of the things it shouldn't

**Comment by mkimberlin** [comment]
Ignore all of the flow-typed/* files...those are generated/downloaded by flow-typed to provide definitions for libraries that aren't built to support flow.

File: package.json [hunk 3]
```diff
@@ -52,6 +53,8 @@
     "jest-cli": "^21.2.1",
     "jest-fetch-mock": "^1.2.1",
     "jest-localstorage-mock": "^2.0.1",
+    "jss": "^9.1.0",
+    "prettier": "^1.7.4",
```
Comment (mkimberlin) [inline package.json:57]: added so that prettier doesn't have to be globally installed

File: package.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
     "classnames": "^2.2.5",
     "color": "^2.0.0",
     "empty": "^0.10.1",
-    "flow-bin": "^0.56.0",
+    "flow-bin": "^0.57.3",
+    "history": "^4.7.2",
```
Comment (mkimberlin) [inline package.json:12]: upgrades and installs to deal with flow errors...

File: src/components/Button/Button.js [hunk 5]
```diff
@@ -1,7 +1,6 @@
 // @flow
 import * as React from "react";
 import cx from "classnames";
-import styleable from "react-styleable";
```
Comment (mkimberlin) [inline src/components/Button/Button.js:4]: unused...

File: src/components/Button/Button.js [hunk 6]
```diff
@@ -20,7 +19,7 @@ type Props = {
  */
 const Button = ({ children, text, className, css = styles }: Props) => (
   <button className={cx("button", css.button, className)}>
-    { (text || children) && <span>{text || children}</span> }
+    {(text || children) && <span>{text || children}</span>}
```
Comment (mkimberlin) [inline src/components/Button/Button.js:22]: prettier updates

File: src/components/Filter/Filter.js [hunk 7]
```diff
@@ -1,20 +1,20 @@
...
+const recfilter = (value, item) =>
+  Object.keys(item).some(
+    k =>
+      k !== "src" && k === "name"
+        ? recfilter(value, item[k].split(" "))
+        : typeof item[k] === "object" || Array.isArray(item[k])
+          ? recfilter(value, item[k])
+          : typeof item[k] === typeof value &&
+            new RegExp(`^${value}`, "i").test(item[k])
+  );
```
Comment (mkimberlin) [inline src/components/Filter/Filter.js:14]: prettier updates...nothing else here.

File: src/components/Filter/Filter.js [hunk 8]
```diff
@@ -1,20 +1,20 @@
 import * as React from "react";
-import { Field, reduxForm } from 'redux-form';
```
Comment (mkimberlin) [inline src/components/Filter/Filter.js:2]: unused

File: src/components/Filter/Filter.js [hunk 9]
```diff
@@ -1,20 +1,20 @@
...
+        ? recfilter(value, item[k].split(" "))
+        : typeof item[k] === "object" || Array.isArray(item[k])
+          ? recfilter(value, item[k])
+          : typeof item[k] === typeof value &&
+            new RegExp(`^${value}`, "i").test(item[k])
+  );
 
-const filter = (data, value) => !value ? data : data.filter(recfilter.bind(null, value));
+const filter = (data, value) =>
+  !value ? data : data.filter(recfilter.bind(null, value));
```
Comment (mkimberlin) [inline src/components/Filter/Filter.js:17]: prettier updates...nothing else here.

File: src/components/Footer/Footer.js [hunk 10]
```diff
@@ -1,6 +1,4 @@
 import * as React from "react";
-import css from "./styles.module.scss";
-import cx from "classnames";
```
Comment (mkimberlin) [inline src/components/Footer/Footer.js:3]: unused

File: src/components/Header/Header.js [hunk 11]
```diff
@@ -3,30 +3,30 @@ import { Link } from "react-router-dom";
...
-      <span className="icon-settings"></span>
+    <Link className={css.settings} to="/preferences">
+      <span className="icon-settings" />
     </Link>
 
     <figure className={cx("image", css.profile_thumbnail)}>
-      <img src="https://avatarfiles.alphacoders.com/696/69632.jpg" alt="Placeholder image" />
+      <img
+        src="https://avatarfiles.alphacoders.com/696/69632.jpg"
+        alt="Placeholder profile thumbnail"
```
Comment (mkimberlin) [inline src/components/Header/Header.js:28]: make eslint happy...

File: src/components/Header/Header.js [hunk 12]
```diff
@@ -55,7 +55,11 @@ const Header = ({
 
         <div className="level-item">
           <Link to="/">
-            <img alt="" className={cx("image", css.logo)} src="/images/logo.svg" />
+            <img
+              alt=""
+              className={cx("image", css.logo)}
+              src="/images/logo.svg"
+            />
```
Comment (mkimberlin) [inline src/components/Header/Header.js:62]: prettier updates...

File: src/components/Icon/Icon.js [hunk 13]
```diff
@@ -18,7 +18,7 @@ const Icon = ({
   size?: string
 }) => {
   return (
-    <span className={cx("icon", { size: sizes[size] }, className)}>
+    <span className={cx("icon", { size: size ? sizes[size]: undefined }, className)}>
```
Comment (mkimberlin) [inline src/components/Icon/Icon.js:21]: flow error fix

File: src/components/Modal/Modal.js [hunk 14]
```diff
@@ -1,21 +1,13 @@
 import * as React from "react";
-import _Modal from "react-modal";
```
Comment (mkimberlin) [inline src/components/Modal/Modal.js:2]: eslint fix...and the rest of this file is prettier updates

File: src/components/Profile/Profile.js [hunk 15]
```diff
@@ -1,27 +1,14 @@
 import * as React from "react";
-import { connect } from 'react-redux';
-import { Link } from 'react-router-dom';
```
Comment (mkimberlin) [inline src/components/Profile/Profile.js:3]: unused...and the rest of the file is prettier updates

File: src/components/Shortcuts/__snapshots__/test.js.snap [hunk 16]
```diff
@@ -6,7 +6,7 @@ exports[`<Shortcuts /> renders without crashing 1`] = `
 >
   <li>
     <ul
-      className="menulist -is-logged-in"
+      className="menulist"
```
Comment (mkimberlin) [inline src/components/Shortcuts/__snapshots__/test.js.snap:9]: this seems to have just been an out of date snapshot

File: src/components/Transactions/Transaction.js [hunk 17]
```diff
@@ -19,21 +12,34 @@ const Transaction = ({
         </div>
       </div>
       <div className="level-item">
-        <img className="user-thumb" src="/images/male_2.jpg" />
+        <img
+          className="user-thumb"
+          src="/images/male_2.jpg"
+          alt="Placeholder avatar thumbnail"
```
Comment (mkimberlin) [inline src/components/Transactions/Transaction.js:18]: make eslint happy...the rest of the file is prettier

File: src/components/Users/User.js [hunk 18]
```diff
@@ -1,22 +1,27 @@
 import * as React from "react";
-import { Link } from "react-router-dom";
```
Comment (mkimberlin) [inline src/components/Users/User.js:2]: unused...the rest of the file is prettier, except the addition of the alt on the img tag

File: src/containers/App/App.js [hunk 19]
```diff
@@ -23,28 +23,30 @@ import Permissions from "routes/Permissions";
 import Preferences from "routes/Preferences";
 import NoMatch from "routes/NoMatch";
 
-import {
-  toggleMenu,
-  closeMenu,
-  routeLoad
-} from "./reducer";
+import { toggleMenu, closeMenu } from "./reducer";
```
Comment (mkimberlin) [inline src/containers/App/App.js:26]: unused routeLoad...most of the rest of this is prettier

File: src/containers/App/App.js [hunk 20]
```diff
@@ -53,43 +55,63 @@ const renderModalRoutes = (props) => (
...
+  isAuthenticated: boolean,
+  handleClickMenu: () => mixed,
+  handleClickMenuClose: () => mixed
+};
+
+type LocationType = {
+  pathname: string,
+  hash: string,
+  search: string
+};
```
Comment (mkimberlin) [inline src/containers/App/App.js:77]: These are some Flow fixes.

File: src/containers/Bootloader/Bootloader.js [hunk 21]
```diff
@@ -5,8 +5,6 @@ import App from "containers/App";
 import Loading from "containers/Loading";
 import { configureStoreAsync } from "util/configureStore";
 
-import store from "fixtures/store";
-
```
Comment (mkimberlin) [inline src/containers/Bootloader/Bootloader.js:9]: unused

File: src/containers/Logout/index.js [hunk 22]
```diff
@@ -1,19 +1,15 @@
...
     dispatch(tryLogout({ history }));
-  },
+  }
 });
 
 class _Logout extends React.Component {
-  constructor(props, context) {
-    super(props, context);
-  }
-
```
Comment (mkimberlin) [inline src/containers/Logout/index.js:16]: eslint fix...useless constructor

File: src/containers/Logout/reducer.js [hunk 23]
```diff
@@ -1,27 +1,25 @@
-const FAIL_LOGOUT = 'FAIL_LOGOUT';
-const SUCCEED_LOGOUT = 'SUCCEED_LOGOUT';
-const TRY_LOGOUT = 'TRY_LOGOUT';
-
-const initialState = {};
```
Comment (mkimberlin) [inline src/containers/Logout/reducer.js:5]: unused...the rest is prettier

File: src/index.js [hunk 24]
```diff
@@ -1,21 +1,25 @@
 // @flow
 import React from "react";
 import ReactDOM from "react-dom";
-import createHistory from "history/createBrowserHistory";
+import { createBrowserHistory as createHistory } from "history";
```
Comment (mkimberlin) [inline src/index.js:4]: flow did not like this at all...not sure why, but this fixed it.

File: src/index.js [hunk 25]
```diff
@@ -1,21 +1,25 @@
...
 import "./styles/index.scss";
 
-const styles = [...document.querySelectorAll('head style')];
+const styles = [...document.querySelectorAll("head style")];
 const link = document.querySelector('link[type="text/css"]');
 
-styles.forEach(style => link.parentElement.insertBefore(style, link));
+styles.forEach(
+  style =>
+    link && link.parentElement && link.parentElement.insertBefore(style, link)
```
Comment (mkimberlin) [inline src/index.js:15]: flow fix...

**Review by ceane — APPROVED** [review]
Changes look good. Approved

**Pull request merged**